tests/cephfs/bug-2053724.py

```diff
@@ -0,0 +1,456 @@
+                else
+                    written += static_cast<size_t>(ret);
+            }
+        }
+
+        CommittedFile::CommittedFile(const std::string& filePath):
+            filePath(filePath)
+        {
+            cleanup();
+        }
+
+        CommittedFile::~CommittedFile()
+        {
+        }
+
+        void CommittedFile::write(const std::string& data)
+        {
+            DirFd dirFd(dirName(filePath));
+            /*
+            * First write and sync work-file. Do not touch real-file.
+            */
+            auto fileName(baseName(filePath));
+            auto workFileName(fileName + ".work");
+            WriteFd workFileFd(dirFd, workFileName);
+            workFileFd.writeAll(data.data(), data.size());
+            workFileFd.sync();
+            workFileFd.close();
+            /**
+            * Posix guarantees that rename is atomic...
+            */
+            dirFd.renameFile(workFileName, fileName);
+            /**
+            * ... and with a directory fsync data is actually stored on disk
+            * See: https://lwn.net/Articles/457667/
+            */
+            dirFd.sync();
+            dirFd.close();
+        }
+
+        std::string CommittedFile::read() const
+        {
+            return readFile(filePath);
+        }
+
+        void CommittedFile::cleanup()
+        {
+            /**
+            * Remove possibly existing old work file
+            */
+            const auto directory(dirName(filePath));
+            DirFd dirFd(directory);
+            const auto fileName(baseName(filePath));
+            const auto workFileName(fileName + ".work");
+            dirFd.unlink(workFileName);
+            dirFd.close();
+        }
+
+        std::string CommittedFile::getPath() const
+        {
+            return filePath;
+        }
+        """
+        test_file = client1.remote_file(
+            sudo=True, file_name="/tmp/test/fsynctest.cpp", file_mode="w"
+        )
+        test_file.write(code)
+        test_file.flush()
+        # test_file.close()
+        client1.exec_command(sudo=True, cmd="yum group install 'Development Tools' -y")
+        client1.exec_command(
+            sudo=True,
+            cmd="g++ -Wall -Wextra -Os /tmp/test/fsynctest.cpp -o /tmp/test/fsynctest",
+        )
+        out, rc = client1.exec_command(
+            sudo=True, cmd=f"/tmp/test/fsynctest {kernel_mounting_dir}/any_file 5"
+        )
+        log.info(f"{out}")
+        output = out.splitlines()
+        log.info(f"{output}")
+        for out in output:
+            line_out = out.split()
+            time = line_out[4]
+            log.info(f"{time}")
+            t = int(time[:-2])
+            log.info(f"time ={t}")
+            if t > 150:
+                raise Exception("Time required is greater than 150ms")
+        log.info("Performance is appropriate")
+        return 0
+
+    except Exception as e:
+        log.info(e)
```

Comment:
Can we make this log.info statements to log.error()

Reply:
Done